server/src/main/java/org/elasticsearch/health/HealthService.java

```diff
@@ -105,24 +108,53 @@ public void getHealth(
                 @Override
                 public void onResponse(FetchHealthInfoCacheAction.Response response) {
                     HealthInfo healthInfo = response.getHealthInfo();
-                    validateResultsAndNotifyListener(
+                    // fork off to the management pool as calculating the indicators can run for longer than is acceptable
+                    // on a transport thread in case of large numbers of indices
+                    ActionRunnable<List<HealthIndicatorResult>> calculateFilteredIndicatorsRunnable = calculateFilteredIndicatorsRunnable(
                         indicatorName,
-                        Stream.concat(filteredPreflightResults, filteredIndicators.map(service -> service.calculate(explain, healthInfo)))
-                            .toList(),
+                        healthInfo,
+                        explain,
                         listener
                     );
+
+                    try {
+                        threadPool.executor(ThreadPool.Names.MANAGEMENT).submit(calculateFilteredIndicatorsRunnable);
```

Comment:
Is management the right thread pool here? Our docs say that's `For cluster management.` I'm not sure this counts as that, does it (or maybe it does -- I could interpret that either way)? Would `generic` be better?

Reply:
I think health it does have to do with the management of the cluster/data/deployment. Generic is quite vague IMO.
Do you have a strong opinion here?

`_cluster/health` goes to `management` too if that matters at all

Reply:
If `_cluster/health` does it, I guess that's precedent enough.